lux-lib/src/luarc/mod.rs

```diff
@@ -0,0 +1,179 @@
+use crate::config::Config;
+use crate::lockfile::LocalPackageLockType;
+use crate::lockfile::ProjectLockfile;
+use crate::lockfile::ReadOnly;
+use crate::project::Project;
+use serde::{Deserialize, Serialize};
+use std::collections::BTreeMap;
+use std::fs;
+use std::path::PathBuf;
+
+#[derive(Serialize, Deserialize, Default, PartialEq, Debug)]
+#[serde(default)]
+struct LuaRC {
+    #[serde(flatten)] // <-- capture any unknown keys here
+    other: BTreeMap<String, serde_json::Value>,
+
+    #[serde(default)]
+    workspace: Workspace,
+}
+
+#[derive(Serialize, Deserialize, Default, PartialEq, Debug)]
+struct Workspace {
+    #[serde(default)]
+    library: Vec<String>,
+}
+
+pub fn update_luarc(config: &Config) {
+    if !config.generate_luarc() {
+        return; // do nothing
+    }
+    let project = Project::current_or_err().expect("failed to get current project");
+    let lockfile = project.lockfile().expect("should have a lockfile");
+    let luarc_path = project.luarc_path();
```

Comment:
suggestion: We probably want to check for both `.luarc.json` and `.emmyrc.json` and then create a `.luarc.json` if neither exists.

Both formats use `workspace.library` fields.

I think for now there's no need to add a `luarc_path` method to `Project`, as it's only used in this module and it's not a core component of a Lux project.